content/java/concepts/hashmap/hashmap.md

```diff
@@ -181,3 +181,35 @@ Jeanette
 Ben
 Lily
 ```
+
+
+##Traversing a `HashMap` using **map.entrySet()**
+
+Alternatively, we could use Map.entry() to return a `set` of key-value pairs. This might be more efficient is some cases, where we might need to access the key and the value.
+```java
+import java.util.HashMap;
+import java.util.entrySet;
+
+public class Main {
+  public static void main(String[] args) {
+    HashMap<String, String> courseTeacher = new HashMap<String, String>();
+
+    courseTeacher.put("History", "Ben");
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+
+
+    for (Map.entry<String, String> entry: courseTeacher.entrySet()) {
+      System.out.printf("Course: %s",entry.getKey());
+      System.out.printf("Teacher: %s \n", entry.getValue());
+    }
+
+  }
+}
+```
+The output will be:
```

Comment:
```suggestion

The output will be:

```

Reply:
does that mean I should delete the line where it says :
"The output will be:"?
I tried to add empty lines before and after, but it still showed me the line in red, until I deleted it completely and it was green and good to go. I'm a bit confused there. I would appreciate any feedback 😃 
Thanks

Reply:
> does that mean I should delete the line where it says : "The output will be:"? I tried to add empty lines before and after, but it still showed me the line in red, until I deleted it completely and it was green and good to go. I'm a bit confused there. I would appreciate any feedback 😃 Thanks

I added the line back. It should have just been blank lines before and after,